x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/action/PostAnalyticsEventAction.java

```diff
@@ -0,0 +1,269 @@
+
+    private PostAnalyticsEventAction() {
+        super(NAME, Response::readFromStreamInput);
+    }
+
+    public static class Request extends ActionRequest implements AnalyticsEvent.Context {
+
+        private final String eventCollectionName;
+
+        private final String eventType;
+
+        private final long eventTime;
+
+        private final boolean debug;
+
+        private final BytesReference payload;
+
+        private final XContentType xContentType;
+
+        public Request(String eventCollectionName, String eventType, boolean debug, XContentType xContentType, BytesReference payload) {
+            this(eventCollectionName, eventType, debug, System.currentTimeMillis(), xContentType, payload);
+        }
+
+        public Request(
+            String eventCollectionName,
+            String eventType,
+            boolean debug,
+            long eventTime,
+            XContentType xContentType,
+            BytesReference payload
+        ) {
+            this.eventCollectionName = eventCollectionName;
+            this.eventType = eventType;
+            this.debug = debug;
+            this.eventTime = eventTime;
+            this.xContentType = xContentType;
+            this.payload = payload;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.eventCollectionName = in.readString();
+            this.eventType = in.readString();
+            this.debug = in.readBoolean();
+            this.eventTime = in.readLong();
+            this.xContentType = in.readEnum(XContentType.class);
+            this.payload = in.readBytesReference();
+        }
+
+        public String eventCollectionName() {
+            return eventCollectionName;
+        }
+
+        public AnalyticsEvent.Type eventType() {
+            return AnalyticsEvent.Type.valueOf(eventType.toUpperCase(Locale.ROOT));
+        }
+
+        public long eventTime() {
+            return eventTime;
+        }
+
+        public BytesReference payload() {
+            return payload;
+        }
+
+        public boolean isDebug() {
+            return debug;
+        }
+
+        public XContentType xContentType() {
+            return xContentType;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (Strings.isNullOrEmpty(eventCollectionName)) {
+                validationException = addValidationError("collection name is missing", validationException);
+            }
+
+            if (Strings.isNullOrEmpty(eventType)) {
+                validationException = addValidationError("event type is missing", validationException);
+            }
+
+            try {
+                if (eventType.toLowerCase(Locale.ROOT).equals(eventType) == false) throw new IllegalArgumentException();
```

Comment:
Should we add a description to the `IllegalArgumentException`?

Reply:
Added a description.